src/index.js

```diff
@@ -241,6 +241,48 @@ class Lock {
       tryAcquire(true);
     });
   }
+
+  /**
+   * Renews a lock in Redis by extending its TTL (time to live).
+   *
+   * @async
+   * @param {Object} lock The lock to renew.
+   * @param {string} lock.id The ID of the lock to renew.
+   * @param {string} lock.index The index of the lock to renew.
+   * @param {number} ttl The new TTL (time to live) for the lock, in milliseconds.
+   *
+   * @returns {Promise<Lock>} A Promise that resolves to an object with the renewed lock's ID, success status, TTL, and index.
+   * @throws {Error} If there is an error executing the renew lock script in Redis.
+   */
+  async renewLock(lock, ttl) {
+    const renewScript = `
+      if redis.call("EXISTS", KEYS[1]) == 0 then
+        return {0, "missing", -1};
+      end;
+
+      local index = redis.call("HGET", KEYS[1], "index");
+      if index ~= ARGV[1] then
```

Comment:
Just curious, why can't we `if redis.call("HGET", KEYS[1], "index") == ARGV[1]` here?

Reply:
We can :) happy to update it with the JDoc suggestion update 👇